Add /info Endpoint (#540)
dotnet/dotnet-monitor: documentation/api/info.md

# Info - Get

Gets information about the Dotnet-Monitor version. Other configuration information may be included in the future.

## HTTP Route

```http
GET /info HTTP/1.1
```

## Host Address

> This route is available on all configured addresses.

Author: Oops, I forgot that everyone would automatically be notified when I pushed - this was just meant to be an intermediary change to save progress since I hadn't started doing any documentation. Sorry about that!

# Info - Get

> Gets information about the Dotnet-Monitor version, the runtime version, and the listening mode.

Reviewer: ```suggestion
Gets information about the `dotnet monitor` version, the runtime version, and the diagnostic port settings.
```

## HTTP Route

```http
GET /info HTTP/1.1
```

## Host Address

The default host address for these routes is `https://localhost:52323`.

## Authentication

Authentication is enforced for this route. See [Authentication](./../authentication.md) for further information.

## Responses

| Name | Type | Description | Content Type |
|---|---|---|---|
| 200 OK | | Information about Dotnet-Monitor formatted as JSON.  | `application/json` |
| 400 Bad Request | [ValidationProblemDetails](definitions.md#ValidationProblemDetails) | An error occurred due to invalid input. The response body describes the specific problem(s). | `application/problem+json` |
| 401 Unauthorized | | Authentication is required to complete the request. See [Authentication](./../authentication.md) for further information. | |

## Examples

### Sample Request

```http

> GET info HTTP/1.1

Reviewer: ```suggestion
GET /info HTTP/1.1
```

Host: localhost:52323
```

### Sample Response

```http
HTTP/1.1 200 OK
Content-Type: application/json

> {"Version":"5.0.0-dev.21362.1+813e40f07bcf3de83b3bb374e2736ad256d0c067","RuntimeVersion":"3.1.16","ListeningMode":"Connect"}

Reviewer: ```suggestion
{
  "Version": "5.0.0-preview.5.21302.5+a99ec90eb220fa3b57f089677d8d1e85d48fb3a8",
  "RuntimeVersion": "3.1.16",
  "DiagnosticPortMode": "Connect"
}
```

we try to line-separate each value for readability. Also, use a shipped version number as the example.

# Info - Get

Gets information about the `dotnet monitor` version, the runtime version, and the diagnostic port settings.

## HTTP Route

```http
GET /info HTTP/1.1
```

## Host Address

The default host address for these routes is `https://localhost:52323`.

## Authentication

Authentication is enforced for this route. See [Authentication](./../authentication.md) for further information.

## Responses

| Name | Type | Description | Content Type |
|---|---|---|---|
| 200 OK | | Information about `dotnet monitor` formatted as JSON.  | `application/json` |
| 400 Bad Request | [ValidationProblemDetails](definitions.md#ValidationProblemDetails) | An error occurred due to invalid input. The response body describes the specific problem(s). | `application/problem+json` |
| 401 Unauthorized | | Authentication is required to complete the request. See [Authentication](./../authentication.md) for further information. | |

## Examples

### Sample Request

```http
GET /info HTTP/1.1
Host: localhost:52323
```

### Sample Response

```http
HTTP/1.1 200 OK
Content-Type: application/json

{

> 	"version":"5.0.0",

Reviewer: nit: Please replace the tabs with four spaces and put a space after each colon for each property.
